pkg/controllers/scheduler/framework/runtime/framework_test.go

```diff
@@ -113,3 +203,112 @@ func TestRunFilterPlugins(t *testing.T) {
+	replicasPlugin := &fakeReplicasPlugin{}
+	filterAndScorePlugin := &fakeFilterAndScorePlugin{}
+	scoreAndSelectPlugin := &fakeScoreAndSelectPlugin{}
+
+	registry := Registry{
+		"filter":         func(f framework.Handle) (framework.Plugin, error) { return filterPlugin, nil },
+		"score":          func(f framework.Handle) (framework.Plugin, error) { return scorePlugin, nil },
+		"select":         func(f framework.Handle) (framework.Plugin, error) { return selectPlugin, nil },
+		"replicas":       func(f framework.Handle) (framework.Plugin, error) { return replicasPlugin, nil },
+		"filterAndScore": func(f framework.Handle) (framework.Plugin, error) { return filterAndScorePlugin, nil },
+		"scoreAndSelect": func(f framework.Handle) (framework.Plugin, error) { return scoreAndSelectPlugin, nil },
+	}
+	tests := []struct {
+		name           string
+		enabledPlugins EnabledPlugins
+		expected       *frameworkImpl
+		shouldError    bool
+	}{
+		{
+			"enabled plugins are respected",
+			EnabledPlugins{
+				FilterPlugins:   []string{"filter"},
+				ScorePlugins:    []string{"score"},
+				SelectPlugins:   []string{"scoreAndSelect"},
+				ReplicasPlugins: []string{"replicas"},
+			},
+			&frameworkImpl{
+				filterPlugins:   []framework.FilterPlugin{filterPlugin},
+				scorePlugins:    []framework.ScorePlugin{scorePlugin},
+				selectPlugins:   []framework.SelectPlugin{scoreAndSelectPlugin},
+				replicasPlugins: []framework.ReplicasPlugin{replicasPlugin},
+			},
+			false,
+		},
+		{
+			"enabled plugins are respected 2",
+			EnabledPlugins{
+				FilterPlugins: []string{"filter", "filterAndScore"},
+				ScorePlugins:  []string{"score", "scoreAndSelect"},
+				SelectPlugins: []string{"scoreAndSelect", "select"},
+			},
+			&frameworkImpl{
+				filterPlugins: []framework.FilterPlugin{filterPlugin, filterAndScorePlugin},
+				scorePlugins:  []framework.ScorePlugin{scorePlugin, scoreAndSelectPlugin},
+				selectPlugins: []framework.SelectPlugin{scoreAndSelectPlugin, selectPlugin},
+			},
+			false,
+		},
+		{
+			"repeated plugins returns error",
+			EnabledPlugins{
+				FilterPlugins:   []string{"filter"},
+				ScorePlugins:    []string{"score", "score"},
+				SelectPlugins:   []string{"scoreAndSelect"},
+				ReplicasPlugins: []string{"replicas"},
+			},
+			nil,
+			true,
+		},
+		{
+			"incorrect type returns error",
+			EnabledPlugins{
+				FilterPlugins:   []string{"replicas"},
+				ScorePlugins:    []string{"score", "scoreAndSelect"},
+				SelectPlugins:   []string{"scoreAndSelect", "select"},
+				ReplicasPlugins: []string{"filter"},
+			},
+			nil,
+			true,
+		},
+		{
+			"plugins not found in registry returns error",
+			EnabledPlugins{
+				FilterPlugins:   []string{"filter"},
+				ScorePlugins:    []string{"score", "scoreAndSelect"},
+				SelectPlugins:   []string{"scoreAndSelect", "select", "notexists"},
+				ReplicasPlugins: []string{"replicas"},
+			},
+			nil,
+			true,
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
Do we need to test that:
- disabled plugins will not be instantiated
- enabled plugins will only be instantiated once?

Reply:
Comments for `PluginConfig` states that 
>  A plugin that is invoked at multiple extension points is initialized once.

These 2 tests are to verify this behavior.
1. If a profile disables a plugin that is present in the registry, we need to ensure its constructor is never called.
2. If a profile enables a plugin at multiple extension points, we need to ensure its constructor is called exactly once.

Reply:
1. `NewFramework` takes in `egistry` and `enabledPlugins`. If a plugin is in `registry` but not in `enabledPlugins`, we should make sure its constructor is never called.  We can do this by registering a plugin that panics on construction, for example. The tests in `profile_test.go` only test the plugin resolution, not the plugin construction.
2. We can also test this using a plugin that panics on repeated construction, for example.

Reply:
got it.

1.  i will add a plugin that panics on construction
2. i guess all plugins should panic on repeated construction?

Reply:
added some checks for 1 and 2